util.go

```diff
@@ -51,11 +58,31 @@ func getExtensionByMimeOrBytes(mime string, bytes []byte) string {
 	return getExtension(bytes)
 }
 
-var unsafeRegex = regexp.MustCompile(`(?i)[^a-z\d+]`)
+// IrcSafeString converts any emoji unicode characters into emojitag, then converts any non-ascii characters into their ascii equivalents, then strips characters that satisfy unsafeRegex, and finally disambiguates the identifier if required
+func IrcSafeString(str string) string {
+	emojiTagged := emoji.UnicodeToEmojiTag(str)
+	decoded := unidecode.Unidecode(emojiTagged)
+	ircSafe := unsafeRegex.ReplaceAllLiteralString(decoded, "")
+	distinct := ensureIdentifierIsDistinct(ircSafe)
+	return distinct
+}
+
+func ensureIdentifierIsDistinct(identity string) string {
+	_, exists := identifiers[identity]
+
+	// we've encountered this identifier before so
+	// increment the counter and append the new count
+	// to the identifier we return
+	if exists {
```

Comment:
```suggestion
	if _, exists := identifiers[identity]; exists {
```

and removing the line from 71-72, of course